music21/romanText/writeRoman.py

```diff
@@ -0,0 +1,552 @@
+
+    if type(beat) in [str, fractions.Fraction]:
+        beat = float(beat)
+
+    # Now beat is definitely a float
+    if int(beat) == beat:
+        return int(beat)
+    else:
+        return round(beat, roundValue)
+
+
+# ------------------------------------------------------------------------------
+
+class Test(unittest.TestCase):
+    '''
+    Tests for two analysis cases (the smallest rntxt files in the music21 corpus)
+    along with two test by modifying those scores.
+
+    Additional tests for the stand alone functions rnString and intBeat and
+    error cases for all three.
+    '''
+
+    def testOpus(self):
+        '''
+        As the rntxt input parser handles Opus objects
+        (i.e. more than one score within the same rntxt files),
+        this RnWriter also needs to accept that type.
+
+        This test parses a fake (tiny) Opus file in three (really tiny!) movements.
+        Checks ensure that the parsed version is indeed an Opus object and that
+        the data is faithfully transferred through that process.
+
+        In practice, Opus handling will bypass this module in the typical case of a simple
+        .write() because writing Opus objects explicitly separates them into their constituent
+        score files prior to invoking this module.
+        '''
+
+        from music21 import converter
+
+        testOpusString = """Composer: Fake composer
+        Piece: Fake piece
+        Movement: 1
+        m1 C: I b3 IV b4 V
+        m2 I
+
+        Movement: 2
+        m1 G: I
+        m3 IV
+        m4 V
+        m5 I
+
+        Movement: 3
+        m1 C: I
+        m2 V
+        m3 I
+        """
+
+        testOpus = converter.parse('romantext: ' + testOpusString)
+        self.assertIsInstance(testOpus, stream.Opus)
+
+        testOpusRnWriter = RnWriter(testOpus)
+        self.assertIn('Title: Fake piece - No.1:', testOpusRnWriter.combinedList)
+        self.assertIn('Title: Fake piece - No.2:', testOpusRnWriter.combinedList)
+        self.assertIn('Title: Fake piece - No.3:', testOpusRnWriter.combinedList)
+        self.assertIn('m2 b1 I', testOpusRnWriter.combinedList)  # mvt 1
+        self.assertIn('m5 b1 I', testOpusRnWriter.combinedList)  # mvt 2
+        self.assertIn('m3 b1 I', testOpusRnWriter.combinedList)  # mvt 3
+
+    def testTwoCorpusPiecesAndTwoCorruptions(self):
+
+        from music21 import corpus
+
+        scoreBach = corpus.parse('bach/choraleAnalyses/riemenschneider004.rntxt')  # Smallest file
+
+        rnaBach = RnWriter(scoreBach)
+        self.assertIn('m10 b1 V6/V b2 V b3 I', rnaBach.combinedList)  # NB b1
+
+        # --------------------
+
+        scoreBach.parts[0].measure(3).insert(0, meter.TimeSignature('10/8'))
+        scoreBach.parts[0].measure(3).insert(1, meter.TimeSignature('5/8'))
+
+        wonkyBach = RnWriter(scoreBach)
+
+        tsString1 = 'Time Signature: 10/8'
+        tsString2 = 'Note: further time signature change(s) unprocessed: [\'5/8\']'
```

Comment:
These are places where it's better to use double quotes on the outside to avoid backslashes internally.